core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/GroupByAggregator.java

```diff
@@ -119,6 +119,7 @@
     put(GroupByConfig.Function.COLLECTLIST, "ARRAY_AGG(%s IGNORE NULLS)");
     put(GroupByConfig.Function.COLLECTSET, "ARRAY_AGG(DISTINCT %s IGNORE NULLS)");
     put(GroupByConfig.Function.COUNTDISTINCT, "COUNT(DISTINCT %s) + IF(COUNTIF(%<s IS NULL) > 0, 1, 0)");
+    put(GroupByConfig.Function.COUNTNULLS, "COUNTIF(%s IS NULL)");
```

Comment:
why do we need this specifically?

Reply:
Because this is one of the aggregations we have listed in the Design doc.